tests/dogtag/pytest-ansible/pytest/performance_test/test_cert_revocation.py

```diff
@@ -0,0 +1,75 @@
+"""
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Description: Load Test client Server Side configuration
+#   Steps :-
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Author: Deepak Punia <dpunia@redhat.com>
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Copyright Red Hat, Inc.
+# SPDX-License-Identifier: GPL-2.0-or-later
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+"""
+from pki.client import PKIConnection
+from pki.cert import CertClient
+import sys
+import logging
+import threading
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--hostname", help="<CA hostname>")
+parser.add_argument("--port", help="<CA Port Number>")
+parser.add_argument("--client-cert", help="path for admin.pem certificate")
+parser.add_argument("--number-of-clients", help="Number of thread", type=int)
+parser.add_argument("--number-of-tests-per-client", help="Number of test per thread", type=int)
+
+log = logging.getLogger()
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+args = parser.parse_args()
+
+# Create a PKIConnection object that stores the details of the CA.
+connection = PKIConnection('https', args.hostname, args.port)
+
+# The pem file used for authentication. Created from a p12 file using the
+# command -
+# openssl pkcs12 -in <p12_file_path> -out /tmp/auth.pem -nodes
+connection.set_authentication_cert(args.client_cert)
+
+# Instantiate the CertClient
+cert_client = CertClient(connection)
+
+
+def run_test(number_of_tests_per_thread):
+    # execute the specified number of tests sequentially
+    # sn > 20 to avoid revocation of initial system certificate
+    for sn in range(20, number_of_tests_per_thread):
+        try:
+            revoke_data = cert_client.revoke_cert(hex(sn), revocation_reason='Key_Compromise',
+                                                  comments="Revoke certificate for testing", nonce=None)
+            log.info("Revoke ID: '{}'".format(revoke_data.request_id))
```

Comment:
We should validate that `revoke_data.operation_result == 'success'`.

Reply:
Done